src/bson.rs

```
@@ -1054,6 +1054,99 @@ impl Binary {
     }
 }
 
+#[cfg(feature = "uuid-0_8")]
+#[cfg_attr(docsrs, doc(cfg(feature = "uuid-0_8")))]
+#[derive(PartialEq)]
+pub enum UuidRepresentation {
+    Standard,
+    CSharpLegacy,
+    JavaLegacy,
+    PythonLegacy,
+}
+
+#[cfg(feature = "uuid-0_8")]
+#[cfg_attr(docsrs, doc(cfg(feature = "uuid-0_8")))]
+impl Binary {
+    pub fn from_uuid(uuid: &uuid::Uuid) -> Self {
+        Binary::from(*uuid)
+    }
+
+    pub fn from_uuid_with_representation(uuid: &uuid::Uuid, rep: UuidRepresentation) -> Self {
+        match rep {
+            UuidRepresentation::Standard => Binary::from_uuid(uuid),
+            UuidRepresentation::CSharpLegacy => {
+                let mut bytes = uuid.as_bytes().to_vec();
+                bytes[0..4].reverse();
+                bytes[4..6].reverse();
+                bytes[6..8].reverse();
+                Binary {
+                    subtype: BinarySubtype::UuidOld,
+                    bytes,
+                }
+            }
+            UuidRepresentation::PythonLegacy => Binary {
+                subtype: BinarySubtype::UuidOld,
+                bytes: uuid.as_bytes().to_vec(),
+            },
+            UuidRepresentation::JavaLegacy => {
+                let mut bytes = uuid.as_bytes().to_vec();
+                bytes[0..8].reverse();
+                bytes[8..16].reverse();
+                Binary {
+                    subtype: BinarySubtype::UuidOld,
+                    bytes,
+                }
+            }
+        }
+    }
+
+    pub fn to_uuid_with_representation(&self, rep: UuidRepresentation) -> Result<uuid::Uuid, &str> {
```

This (and others) should return an error enum rather than a &str.